boot/bootutil/src/loader.c

```diff
@@ -2173,21 +2188,32 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
                          BOOT_IMG(state, BOOT_PRIMARY_SLOT).hdr.ih_magic,
                          BOOT_CURR_IMG(state));
             rc = BOOT_EBADIMAGE;
+            FIH_SET(fih_rc, FIH_FAILURE);
             goto out;
         }
 #endif /* MCUBOOT_VALIDATE_PRIMARY_SLOT */
 
         rc = boot_update_hw_rollback_protection(state);
         if (rc != 0) {
+            FIH_SET(fih_rc, FIH_FAILURE);
             goto out;
         }
 
         rc = boot_add_shared_data(state, BOOT_PRIMARY_SLOT);
         if (rc != 0) {
+            FIH_SET(fih_rc, FIH_FAILURE);
             goto out;
         }
+        ++fih_cnt;
+    }
+    /*
+     * fih_cnt should be equal to BOOT_IMAGE_NUMBER now. 
```

Comment:
It doesn't show here, but this line and the next have a trailing space. Please fix that.